support/Pipelines/Bbh/InitialData.py

```diff
@@ -0,0 +1,143 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+from pathlib import Path
+from typing import Union
+
+import click
+from rich.pretty import pretty_repr
+
+from spectre.support.Schedule import schedule, scheduler_options
+
+logger = logging.getLogger(__name__)
+
+ID_INPUT_FILE_TEMPLATE = Path(__file__).parent / "InitialData.yaml"
+
+
+def id_parameters(
+    mass_ratio: float,
+    separation: float,
+    orbital_angular_velocity: float,
+    refinement_level: int,
+    polynomial_order: int,
+):
+    """Determine initial data parameters from options.
+
+    These parameters fill the 'ID_INPUT_FILE_TEMPLATE'.
+
+    Arguments:
+      mass_ratio: Defined as q = M_A / M_B >= 1.
+      separation: Coordinate separation D of the black holes.
+      orbital_angular_velocity: Omega_0.
+      refinement_level: h-refinement level.
+      polynomial_order: p-refinement level.
+    """
+
+    # Sanity checks
+    assert mass_ratio >= 1.0, "Mass ratio is defined to be >= 1.0."
+
+    # Determine initial data parameters from options
+    M_A = mass_ratio / (1.0 + mass_ratio)
+    M_B = 1.0 / (1.0 + mass_ratio)
+    x_A = separation / (1.0 + mass_ratio)
+    x_B = x_A - separation
+    return {
+        "MassRight": M_A,
+        "MassLeft": M_B,
+        "XRight": x_A,
+        "XLeft": x_B,
+        "ExcisionRadiusRight": 0.89 * 2.0 * M_A,
+        "ExcisionRadiusLeft": 0.89 * 2.0 * M_B,
+        "OrbitalAngularVelocity": orbital_angular_velocity,
+        # Resolution
+        "L": refinement_level,
+        "P": polynomial_order,
+    }
+
+
+def generate_id(
+    mass_ratio: float,
+    separation: float,
+    orbital_angular_velocity: float,
+    refinement_level: int,
+    polynomial_order: int,
+    id_input_file_template: Union[str, Path] = ID_INPUT_FILE_TEMPLATE,
+    **scheduler_kwargs,
+):
+    """Generate initial data for a BBH simulation.
+
+    Parameters for the initial data will be inserted into the
+    'id_input_file_template'. The remaining options are forwarded to the
+    'schedule' command. See 'schedule' docs for details.
+    """
+    # Determine initial data parameters from options
+    id_params = id_parameters(
+        mass_ratio=mass_ratio,
+        separation=separation,
+        orbital_angular_velocity=orbital_angular_velocity,
+        refinement_level=refinement_level,
+        polynomial_order=polynomial_order,
+    )
+    logger.debug(f"Initial data parameters: {pretty_repr(id_params)}")
+
+    # Schedule!
+    return schedule(id_input_file_template, **id_params, **scheduler_kwargs)
+
+
+@click.command(name="generate-id", help=generate_id.__doc__)
+@click.option(
+    "--mass-ratio",
```

Comment:
How do you feel about using another yaml to specify these? As we add more and more capability, having to write a new click option for every nob we want to tweak will probably get pretty annoying. This could simplify the command down to

```sh
spectre bbh generate-id InitialParams.yaml
```

where `InitialParams.yaml` is

```yaml
mass-ratio: 1.5
separation: 16
orbital-angular-velocity: 0.01
polynomial-order: 7
```

and then all the other parameters we didn't specify will be default.

This suggestion applies to all phases of the BBH pipeline.

Reply:
Ohh no, adding another "meta" input file seems like it could be a disaster :D I'd rather keep the knobs here down, and even remove the hp refinement options once we have AMR. If you want full control over options you can write your own input file and run it. The steps in the pipeline still work if you're starting them from your own simulation that you didn't run with the pipeline.

Reply:
I would be fine with adding an option to supply the CLI option from a configuration file rather than on the command line if you want. We  already do something like that with the user config file and the status CLI. It's just an alternative way of passing the options to the CLI. There's a pretty generic way to do that with click.

Reply:
Did you want to make this change now or in a follow up PR?

Reply:
I wasn't planning to do this right away, since I don't have a use case right now

Reply:
I agree, could be nice, first need to see how all this works in practice :)